src/mysql_conn.erl

```diff
@@ -553,10 +553,22 @@ handle_info(ping, #state{socket = Socket, sockmod = SockMod} = State) ->
     #ok{} = mysql_protocol:ping(SockMod, Socket),
     setopts(SockMod, Socket, [{active, once}]),
     {noreply, schedule_ping(State)};
-handle_info({tcp_closed, _Socket}, State) ->
+handle_info({tcp_closed, Socket}, #state{socket = Socket} = State) ->
     {stop, normal, State#state{socket = undefined, connection_id = undefined}}; 
-handle_info({tcp_error, _Socket, Reason}, State) ->
+handle_info({ssl_closed, Socket}, #state{socket = Socket} = State) ->
+    {stop, normal, State#state{socket = undefined, connection_id = undefined}}; 
+handle_info({tcp_error, Socket, Reason}, #state{socket = Socket} = State) ->
     stop_server({tcp_error, Reason}, State);
+handle_info({ssl_error, Socket, Reason}, #state{socket = Socket} = State) ->
+    stop_server({ssl_error, Reason}, State);
+handle_info({tcp, Socket, _Data}, #state{socket = Socket,
+                                         sockmod = SockMod} = State) ->
+    setopts(SockMod, Socket, [{active, once}]),
+    {noreply, State};
+handle_info({ssl, Socket, _Data}, #state{socket = Socket,
+                                         sockmod = SockMod} = State) ->
+    setopts(SockMod, Socket, [{active, once}]),
+    {noreply, State};
```

Comment:
A comment that this is data spontaneously pushed by the server could help understanding this.

We could parse it an see that it is the "I'm going to close this connection" event, but I don't know if it's necessary.

Reply:
Comments added.